internal/cephfs/store/fsjournal.go

```diff
@@ -128,7 +128,20 @@ func CheckVolExists(ctx context.Context,
 		}
 		err = cloneState.ToError()
 		if errors.Is(err, cerrors.ErrCloneInProgress) {
-			return nil, err
+			progressReport := cloneState.GetProgressReport()
+			if progressReport.PercentageCloned == "" || progressReport.AmountCloned == "" ||
+				progressReport.FilesCloned == "" {
+				return nil, err
+			} else {
+				err = fmt.Errorf("%w. progress report: percentage cloned=%s, amount cloned=%s, files cloned=%s",
+					err,
+					progressReport.PercentageCloned,
+					progressReport.AmountCloned,
+					progressReport.FilesCloned)
+				log.UsefulLog(ctx, err.Error())
+
+				return nil, err
```

Comment:
```suggestion
if progressReport.PercentageCloned == "" && progressReport.AmountCloned == "" &&
  progressReport.FilesCloned == "" {
     return nil, err
}
msg:="progress report:"
if progressReport.PercentageCloned != "" {
msg=fmt.Sprintf("%s percentage cloned=%s,progressReport.PercentageCloned)
}
if  progressReport.AmountCloned != "" {
msg=fmt.Sprintf("%s amount cloned=%s,progressReport.AmountCloned)
}
if progressReport.FilesCloned != "" {
msg=fmt.Sprintf("%s files cloned=%s=%s,progressReport.FilesCloned)
}
err = fmt.Errorf("%w. msg)
log.UsefulLog(ctx, err.Error())

return nil, err
```

can we do like this?

Reply:
I doubt that only one of them would ever be reported, it will either be all or none.
Even if the values are empty it would be "0" and not empty. So, I don't think we need something elaborative like this.

Reply:
we need to be sure on it , because its a string not int to show `0`,

Reply:
Yes, I have verified it. Even if the values are 0 (i.e, for empty directories, ), the output will be 0 wrapped in string.
Ref: https://github.com/ceph/ceph/blob/main/src/pybind/mgr/volumes/fs/stats_util.py#L26-L74